### ISSUE TITLE
Should a `TcpConnection` provide a `close`?

### ISSUE THREAD

**Issue by NiteshKant** [reactive-ipc/reactive-ipc-jvm#9]
Currently a [`Connection`](https://github.com/reactive-ipc/reactive-ipc-jvm/blob/master/ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/Connection.java#L9) is a `Publisher<T>` which essentially means that an unsubscription from this connection would result in a close.

The issue with this approach is that for consumers that do not want to consume the content of the connection have no easy way to close the connection as typically they will not subscribe to the content.

**Comment by jbrisbin** [comment]
Isn't that what the `Subscription.cancel()` method is for?

**Comment by NiteshKant** [comment]
> > Isn't that what the Subscription.cancel() method is for?

Yes. Although, the case in point here is to support a usecase of `ignoreContent` where in the receiver is not interested in the incoming content at all and hence will never subscribe.
